### PR TITLE
Add @Rest parameter to inject ResponseErrorHandler

### PR DESCRIPTION
**Pull request by simopete**
Implements #926

### CODE REVIEW

**Comment by simopete** [comment]
I also added it to Rest-API wiki. It is in my wiki repo on 926_ResponseErrorHandler_wiki branch.

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java [hunk 1]
```diff
@@ -170,6 +170,47 @@
  * </pre>
  * 
  * </blockquote>
+ *
+ * <h1>ResponseErrorHandler</h1>
+ * <p>
+ * You can use your own error handler if you want to customize how error
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java]: how error**s**

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java [hunk 2]
```diff
@@ -170,6 +170,47 @@
...
+ * are handled. The {@link #responseErrorHandler()} parameter lets you define the
+ * {@link org.springframework.web.client.ResponseErrorHandler
+ * ResponseErrorHandler}.
+ * </p>
+ *
+ * <p>
+ * You can inject {@link org.androidannotations.annotations.EBean EBean} response error
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java]: **a** response error

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java [hunk 3]
```diff
@@ -170,6 +170,47 @@
...
+ * handler just like as request factory.
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java]: as **a** request factory

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java [hunk 4]
```diff
@@ -170,6 +170,47 @@
  * </pre>
  * 
  * </blockquote>
+ *
+ * <h1>ResponseErrorHandler</h1>
+ * <p>
+ * You can use your own error handler if you want to customize how errors
+ * are handled. The {@link #responseErrorHandler()} parameter lets you define the
+ * {@link org.springframework.web.client.ResponseErrorHandler
+ * ResponseErrorHandler}.
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java]: You can use your own error handler to customize how errors
+ * are handled. The {@link #responseErrorHandler()} parameter lets you define the
+ * {@link org.springframework.web.client.ResponseErrorHandler
+ * ResponseErrorHandler}.

File: AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java [hunk 5]
```diff
@@ -170,6 +170,47 @@
...
+ * </p>
+ *
+ * <p>
+ * You can inject {@link org.androidannotations.annotations.EBean EBean} a response error
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-rest-spring/rest-spring-api/src/main/java/org/androidannotations/rest/spring/annotations/Rest.java]: You can inject an {@link org.androidannotations.annotations.EBean EBean} response error
+ * handler just like as a request factory.

**Comment by WonderCsabo** [comment]
Merged, thanks!

**Pull request merged**